### ISSUE TITLE
no-console warns on assignment

### ISSUE THREAD

**Issue by sachinraja** [eslint/eslint#14793]
**Tell us about your environment**

* **ESLint Version:** v7.30.0
* **Node Version:** v16.4.0
* **npm Version:** v7.18.1
* **Operating System:** linux 5.8.0-58-generic

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
      ecmaVersion: 6
  },
  rules: {
    "no-console": "warn",
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.log = jest.fn()
```

```bash
npx eslint .
```

**What did you expect to happen?**
The `no-console` rule should detect that this is an assignment and not a call. In [the rule details](https://eslint.org/docs/rules/no-console#rule-details) it says "This rule disallows calls to methods of the console object."

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
`warning  Unexpected console statement  no-console`

**Steps to reproduce this issue:**

1. in your eslint configuration add `no-console: 'warn'` to your `rules`.
2. in your code, set `console.log = null`

**Are you willing to submit a pull request to fix this bug?**
yes

**Comment by mdjermanovic** [comment]
Hi @sachinraja, thanks for the issue!

> console.log = jest.fn()

This was already discussed in https://github.com/eslint/eslint/issues/7806, and it was eventually decided that this behavior shouldn't be changed.

If you want to disallow only method calls, you can use the `no-restricted-syntax` rule, as described in [When Not To Use It](https://eslint.org/docs/rules/no-console#when-not-to-use-it) section of `no-console` docs ("Another case where you might not use this rule is if you want to enforce console calls and not console overwrites...").

> In [the rule details](https://eslint.org/docs/rules/no-console#rule-details) it says "This rule disallows calls to methods of the console object."

You're right, this sounds misleading. I'm marking this issue as an accepted documentation change.

**Comment by coderaiser** [comment]
@sachinraja you can use [@putout/plugin-remove-console](https://github.com/coderaiser/putout/tree/v18.4.0/packages/plugin-remove-console) rule of [putout](https://github.com/coderaiser/putout) with help of [eslint-plugin-putout](https://github.com/coderaiser/putout/tree/v18.4.0/packages/eslint-plugin-putout). It works in the way you need it.
